client.go

```diff
@@ -98,7 +98,10 @@ func decodeString(body io.Reader, output *string) error {
 	return nil
 }
 
-func (c *Client) fullURL(suffix string) string {
+// fullURL returns full URL for request.
+// If API type is Azure, model is required to get deployment name.
+// If API type is OpenAI, model will be ignored.
+func (c *Client) fullURL(suffix, model string) string {
```

Comment:
As a suggestion: do we want to use variadic options here instead of the required `model` argument? That would require fewer changes to other parts of the library. https://gobyexample.com/variadic-functions

We didn't adopt this pattern for the whole library in #278, but it seems useful in the context of the `fullURL` method.

What do you think?

Reply:
I agree, I would be happy to do this job.